Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -15,16 +15,19 @@
 
 package com.amazon.randomcutforest.sampler;
 
-import static com.amazon.randomcutforest.CommonUtils.checkNotNull;
-import static com.amazon.randomcutforest.CommonUtils.checkState;
-
 import java.util.List;
 import java.util.Optional;
 import java.util.Random;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 import java.util.stream.Stream;
 
+import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+import static com.amazon.randomcutforest.CommonUtils.checkState;
+import static com.amazon.randomcutforest.RandomCutForest.DEFAULT_INITIAL_ACCEPT_FRACTION;
+import static com.amazon.randomcutforest.RandomCutForest.DEFAULT_SAMPLE_SIZE;
+import static com.amazon.randomcutforest.RandomCutForest.DEFAULT_STORE_SEQUENCE_INDEXES_ENABLED;
```

Comment:
If we need to have default values both here and in RandomCutForest, then I think the dependency should go the other way: RandomCutForest should import that constants from this class. That said, I don't think CompactSampler should have a default sample size, because having a default sample size makes sense in the context of a Random Cut Forest, but not for a sampler in general. Similarly, if a default initial accept fraction is set in this class, then I think it should either be 0 or 1. Any other choice seems like it would be application dependent, and hence doesn't belong in the sampler class.

Alternatively, we don't have to set defaults in this class at all, and we can require that the parameters be set explicitly when creating a sampler.

Reply:
I'm ok either way. Maybe clean up all the builder related options in one go (there will be multiple classes with these kinds of issues).

Reply:
Sure, can you create an issue for tracking?